plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/impl/SynchronizedEObjectWithInverseResolvingEList.java

```diff
@@ -118,11 +121,31 @@ public int hashCode() {
         }
     }
 
+    @Override
+    public Object[] toArray() {
+        lock.lock();
+        try {
+            return super.toArray();
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    @Override
+    public <T> T[] toArray(T[] array) {
+        lock.lock();
+        try {
+            return super.toArray(array);
+        } finally {
+            lock.unlock();
+        }
+    }
+
```

Comment:
Please prepare a seperate pull request for this change. my read: it fixes problems as follows and uses the monitor to synchronize:

```
Caused by: org.eclipse.emf.common.util.BasicEList$BasicIndexOutOfBoundsException: index=13, size=13
	at org.eclipse.emf.common.util.BasicEList.get(BasicEList.java:352)
	at org.locationtech.udig.project.internal.impl.SynchronizedEObjectWithInverseResolvingEList.get(SynchronizedEObjectWithInverseResolvingEList.java:78)
	at org.eclipse.emf.ecore.util.EcoreEList.toArray(EcoreEList.java:222)
	at java.util.Collections$UnmodifiableCollection.toArray(Unknown Source)
	at org.eclipse.emf.edit.ui.provider.AdapterFactoryContentProvider.getElements(AdapterFactoryContentProvider.java:161)
	at org.eclipse.jface.viewers.StructuredViewer.getRawChildren(StructuredViewer.java:989)
	at org.eclipse.jface.viewers.ColumnViewer.getRawChildren(ColumnViewer.java:703)
	at org.eclipse.jface.viewers.AbstractTreeViewer.getRawChildren(AbstractTreeViewer.java:1332)
	at org.eclipse.jface.viewers.TreeViewer.getRawChildren(TreeViewer.java:391)
	at org.eclipse.jface.viewers.StructuredViewer.getFilteredChildren(StructuredViewer.java:896)
	at org.eclipse.jface.viewers.AbstractTreeViewer.getSortedChildren(AbstractTreeViewer.java:601)
	at org.eclipse.jface.viewers.AbstractTreeViewer.updateChildren(AbstractTreeViewer.java:2581)
```

Reply:
See https://github.com/locationtech/udig-platform/pull/651